public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -86,6 +86,17 @@ const featuresToTempoVersion = {
 // This is the last minor version of Tempo that does not expose the endpoint for build information.
 const defaultTempoVersion = '2.1.0';
 
+interface ServiceMapQueryResponse {
+  nodes: DataFrame;
+  edges: DataFrame;
+}
+
+interface ServiceMapQueryResponseWithRates {
+  rates: Array<DataFrame | DataFrameDTO>;
+  nodes: DataFrame;
+  edges: DataFrame;
+}
```

Comment:
So this is the biggest change here to help make the code a bit more understandable and fix type errors.  

These two types are now used for the return type of serviceMapQuery and rateQuery instead of using a single data array that by the ended up looking like this
`data: Array<Array<DataQueryResponseData>, DataQueryResponseData, DataQueryResponseData>`

DataQueryResponseData resolves to  `DataFrame | DataFrameDTO`  (in main it also resolves to any, which I am trying to solve)